app/models/transaction_type.rb

```diff
@@ -0,0 +1,29 @@
+class TransactionType < ApplicationRecord
+  NAMES = {
+    credit: %i[
+      salary benefits
+      maintenance_in
+      property_or_lodger
+      student_loan
+      pension
+      fiends_or_family
+    ],
+    debit: %i[
+      rent_or_mortgage
+      council_tax
+      child_care
+      maintenance_out
+      legal_aid
+    ]
+  }.freeze
+
+  scope :debits, -> { where(operation: :debit) }
+  scope :credits, -> { where(operation: :credit) }
+
+  def self.populate
+    existing = pluck(:name).map(&:to_sym)
+    NAMES.each do |operation, names|
+      (names - existing).each { |name| create!(name: name, operation: operation) }
```

Comment:
you could also use `.find_or_create!` instead of fetching the existing ones. Just a suggestion. Feel free to ignore

Reply:
The problem with using `find_or_create` in the loop is that it will make a SQL call (to see if one exists) on each iteration of the loop. Doing the `pluck` means only SQL to determine existing items. And the only SQL made after that being to create the items that don't exist.

Reply:
Ok it makes sense